### PR TITLE
Feature/point selection

### PR DESCRIPTION
**Pull request by jacob-andersson**
Checkout demo see how it works (Tap and hold for a while, then start to pan)

### CODE REVIEW

File: Source/HYPScatterPlot.m [hunk 1]
```diff
@@ -74,107 +80,109 @@ - (instancetype)initWithFrame:(CGRect)frame
...
                               @"plotView": self.plotView};
-
+    
     [self addConstraints:
      [NSLayoutConstraint constraintsWithVisualFormat:@"H:|-padding-[minimumXLabel]"
                                              options:0
                                              metrics:metrics
                                                views:views]];
-
+
```
Comment (3lvis) [inline Source/HYPScatterPlot.m]: Ewwww, have you tried https://github.com/zenangst/MarvinXcode? It trims whitespaces like this ones

Comment (jacob-andersson) [inline Source/HYPScatterPlot.m]: Yeah, I'm using it, strange that it did not apply it's magic in this case :(

Comment (3lvis) [inline Source/HYPScatterPlot.m]: Ping @zenangst :scream:

Comment (attila-at-hyper) [inline Source/HYPScatterPlot.m]: @jacob-andersson haven't you recently upgraded Xcode? that might break existing plug-ins

Comment (jacob-andersson) [inline Source/HYPScatterPlot.m]: @attila-at-hyper Could be

**Comment by jacob-andersson** [comment]
Yey, reboot xcode did the trick :dancers:

File: Source/HYPScatterPlot.m (hunk 1 above)
Comment (zenangst) [inline Source/HYPScatterPlot.m]: tumblr_lq0kcuky0s1r1rvcno1_500

**Comment by 3lvis** [comment]
ios simulator screen shot apr 8 2015 3 12 59 pm

**Comment by 3lvis** [comment]
@jacob-andersson Should the dot only change color? Or is it supposed to show its value too.

**Comment by 3lvis** [comment]
dwqdqw

**Comment by zenangst** [comment]
@jacob-andersson have a look at https://github.com/hyperoslo/Form/blob/master/Source/FORMTooltipView.m

It pretty much does what @3lvis has in his mockup.

**Comment by jacob-andersson** [comment]
@3lvis @zenangst: I guess that would be a nice to have. When using Scatter at the moment I can however see the need to heavily customize the "tooltip" to be shown outside of the actual plot view, also providing additional description for the presented value etc. Perhaps this additional feature can be included in a later release?

**Comment by 3lvis** [comment]
What's the point on showing the dot if I don't know its value?

Or do you have a alternative way to show the current highlighted value?

**Comment by jacob-andersson** [comment]
The idea (for now at least) is that the coordinates of the point is provided in the delegate method so that one can choose to display whatever label. Also the selection altogether is optional with the enableSelection property, and you can also choose fill and stroke color + vertical line color using properties. 

But as I pointed out, YES, it would be a nice future addition!

File: Source/HYPScatterPlot.m [hunk 2]
```diff
@@ -34,6 +34,10 @@ @interface HYPScatterPlot()
 
 @property (nonatomic) UIView *plotView;
 
+@property (nonatomic) UILongPressGestureRecognizer *panGestureRecognizer;
```
Comment (3lvis) [inline Source/HYPScatterPlot.m]: panGestureRecognizer != longPressGestureRecognizer

You might want to reconsider the naming of this variable

File: Source/HYPScatterPlot.m [hunk 3]
```diff
@@ -506,29 +552,56 @@ - (void)drawPointsAndYThresholdsInRect:(CGRect)rect
...
+    CGPoint selectedPoint = CGPointMake(CGFLOAT_MAX, CGFLOAT_MAX);
+    NSUInteger selectedIndex;
+
     for (NSInteger index = 0; index < numberOfPoints; index++) {
         CGPoint point = [self.dataSource pointAtIndex:index];
         CGPoint translatedPoint = CGPointMake(
-            [self translateValue:point.x srcMin:srcMinX srcMax:srcMaxX destMin:destMinX destMax:destMaxX],
-            [self translateValue:point.y srcMin:srcMinY srcMax:srcMaxY destMin:destMinY destMax:destMaxY]
-        );
+                                              [self translateValue:point.x srcMin:srcMinX srcMax:srcMaxX destMin:destMinX destMax:destMaxX],
```
Comment (3lvis) [inline Source/HYPScatterPlot.m]: Mind dropping the abbreviations here?
-                                              [self translateValue:point.x srcMin:srcMinX srcMax:srcMaxX destMin:destMinX destMax:destMaxX],

File: Source/HYPScatterPlot.m [hunk 4]
```diff
@@ -506,29 +552,56 @@ - (void)drawPointsAndYThresholdsInRect:(CGRect)rect
...
+                                              [self translateValue:point.y srcMin:srcMinY srcMax:srcMaxY destMin:destMinY destMax:destMaxY]
+                                              );
+
+        [self drawPoint:translatedPoint isSelected:NO];
+
+        if (self.enableSelection && self.selectedX != CGFLOAT_MAX && fabsf(translatedPoint.x - self.selectedX) < fabsf(selectedPoint.x - self.selectedX)) {
```
Comment (3lvis) [inline Source/HYPScatterPlot.m]: Newlines after && or ||

``` objc
if (self.enableSelection && 
    self.selectedX != CGFLOAT_MAX && 
    fabsf(translatedPoint.x - self.selectedX) < fabsf(selectedPoint.x - self.selectedX))
```

And even better if you wrapped on what's the intention behind this, for example:

``` objc
BOOL supportsSelection = (self.enableSelection && 
                          self.selectedX != CGFLOAT_MAX && 
                          fabsf(translatedPoint.x - self.selectedX) < fabsf(selectedPoint.x - self.selectedX));
if (suppportsSelection) {
```

(check other places for a similar improvement)

**Comment by 3lvis** [comment]
@jacob-andersson I understand your approach, not my favorite one but I understand it works.

File: Source/HYPScatterPlot.m (hunk 2 above)
Comment (jacob-andersson) [inline Source/HYPScatterPlot.m]: @3lvis It is the same gesture recognizer for both the initial long pressure and the panning, have any genius suggestion for namning?

File: Source/HYPScatterPlot.m (hunk 3 above)
Comment (jacob-andersson) [inline Source/HYPScatterPlot.m]: @3lvis I might be a bit slow, but care to elaborate on that?

File: Source/HYPScatterPlot.m (hunk 2 above)
Comment (3lvis) [inline Source/HYPScatterPlot.m]: @jacob-andersson What about `gestureRecognizer`

File: Source/HYPScatterPlot.m (hunk 3 above)
Comment (3lvis) [inline Source/HYPScatterPlot.m]: destMin = destinationMinimum
srcMax = sourceMaximum
destMaxX = destinationMaximumX

**Comment by 3lvis** [comment]
Awesome!

**Pull request merged**